ra2ce/analysis/indirect/multi_link_origin_closest_destination.py

```diff
@@ -0,0 +1,187 @@
+            (
+                base_graph,
+                opt_routes_without_hazard,
+                destinations,
+            ) = analyzer.optimal_route_origin_closest_destination()
+
+            if self.graph_file_hazard.file is None:
+                origins = analyzer.load_origins()
+                opt_routes_with_hazard = GeoDataFrame(data=None)
+            else:
+                (
+                    base_graph,
+                    origins,
+                    destinations,
+                    agg_results,
+                    opt_routes_with_hazard,
+                ) = analyzer.multi_link_origin_closest_destination()
+
+                (opt_routes_with_hazard) = (
+                    analyzer.difference_length_with_without_hazard(
+                        opt_routes_with_hazard, opt_routes_without_hazard
+                    )
+                )
+        else:
+            (
+                base_graph,
+                origins,
+                destinations,
+                agg_results,
+                opt_routes_with_hazard,
+            ) = analyzer.multi_link_origin_closest_destination()
+            opt_routes_without_hazard = GeoDataFrame()
+
+        if self.analysis.save_gpkg:
+            # Save the GeoDataFrames
+            to_save_gdf = [
+                origins,
+                destinations,
+                opt_routes_without_hazard,
+                opt_routes_with_hazard,
+            ]
+            to_save_gdf_names = [
+                "origins",
+                "destinations",
+                "optimal_routes_without_hazard",
+                "optimal_routes_with_hazard",
+            ]
+            _save_gpkg_analysis(base_graph, to_save_gdf, to_save_gdf_names)
+        if self.analysis.save_csv:
+            csv_path = _output_path.joinpath(
+                self.analysis.name.replace(" ", "_") + "_destinations.csv"
+            )
+            if "geometry" in destinations.columns:
+                del destinations["geometry"]
+            if not csv_path.parent.exists():
+                csv_path.parent.mkdir(parents=True)
+            destinations.to_csv(csv_path, index=False)
+
+            csv_path = _output_path.joinpath(
+                self.analysis.name.replace(" ", "_") + "_optimal_routes.csv"
+            )
+            if not opt_routes_without_hazard.empty:
+                del opt_routes_without_hazard["geometry"]
+                opt_routes_without_hazard.to_csv(csv_path, index=False)
+            if not opt_routes_with_hazard.empty:
+                del opt_routes_with_hazard["geometry"]
+                opt_routes_with_hazard.to_csv(csv_path, index=False)
+
+        if self.graph_file_hazard.file is not None:
+            agg_results.to_excel(
+                _output_path.joinpath(
+                    self.analysis.name.replace(" ", "_") + "_results.xlsx"
+                ),
+                index=False,
+            )
+
+        return None
```

Comment:
Return None?

Reply:
Correct, not all analyses return a GeoDataFrame